src/CarbonAware.DataSources/CarbonAware.DataSources.Json/src/Configuration/ServiceCollectionExtensions.cs

```diff
@@ -14,6 +14,8 @@ public static void AddJsonDataSourceService(this IServiceCollection services, IC
         {
             configuration.GetSection(JsonDataSourceConfiguration.Key).Bind(c);
         });
-        services.TryAddSingleton<ICarbonIntensityDataSource, JsonDataSource>();
+        services.TryAddSingleton<IForecastDataSource, JsonDataSource>();
```

Comment:
Shouldn't be JSON type not implementing `IForecastDataSource`? That would be insync with the proposal of removing those `Not Implemented` exception cases.

Reply:
This is part 1 of the datasource configuration change where we just separated the data source interfaces. The followup PRs - [188 ](https://github.com/Green-Software-Foundation/carbon-aware-sdk/pull/188) and [190](https://github.com/Green-Software-Foundation/carbon-aware-sdk/pull/190) will cover the entire config schema changes and implementation